src/py42/response.py

```diff
@@ -22,6 +22,9 @@ def __init__(self, requests_response):
     def __getitem__(self, key):
         return self._data_root[key]
 
+    def __setitem__(self, key, value):
+        self._data_root[key] = value
```

Comment:
Should we add a check here to make sure `_data_root` isn't a string, and if it is print a helpful error? Maybe something like "Response data of type str does not support item assignment"?

Reply:
python will make something like happen automatically without us doing anything:
```
>>> test = "test"
>>> test[0] = 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'str' object does not support item assignment
>>> test["test"]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: string indices must be integers, not str
```

Reply:
yeah, I'm just thinking about the fact that since we're doing some misdirection on the __setitem__ it could be confusing to debug because the types don't match. 

I could see an end user saying: "what str object? I tried to set a value on a Py42Response object"

Reply:
shouldn't the trace help with that?

Reply:
I suppose we could catch/raise this in a `Py42Error` and add some additional context